File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n);
        int left = 0, right = n - 1;
        
        for (int i = n - 1; i >= 0; --i){
            int sq;
            if (abs(nums[left]) < abs(nums[right])) sq = nums[right--];
            else sq = nums[left++];
            
            res[i] = sq * sq;
        }
    return res;
    }
};

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        n = len(A)
        res = [0] * n
        left, right = 0, n - 1
        
        for i in range(n - 1, -1, -1):
            if abs(A[right]) > abs(A[left]):
                sq = A[right]
                right -= 1
            else:
                sq = A[left]
                left += 1
            res[i] = sq * sq
        return res
        